### PR TITLE
Remove cycles from CMake dependency graph

### PR DESCRIPTION
**Pull request by bracketttc**
Attempts to generate a Ninja build from the existing CMake structure failed complaining about phony cycles related to the use of `add_custom_target` and placing the primary output file in the `DEPENDS` option.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -434,17 +434,17 @@ function(add_tarball targetname namever treeish)
 	set(distname ${tarname}.bz2)
 	set(docname ${namever}-doc.${distfmt})
 
-	add_custom_target(${docname}
-		DEPENDS man apidoc
+	file(GLOB man_pages docs/man/*.[1-8])
```
Comment (pmatilai) [inline CMakeLists.txt:437]: What's the reason for this? AFAICS the tarball gets updated on man page changes as it is, through the "man" target.

**Comment by pmatilai** [comment]
Right, I remember encountering this pattern somewhere. Always looked like one of those cmake WTFs for me, which is probably why I tried to find some other way to do it. And as such, it was probably always for the worse.

I'll accept changing this for correctness, but note that we officially only support "make" based builds as per INSTALL.

**Review by pmatilai — CHANGES_REQUESTED** [review]
I came *this* close to merging, but testing made me remember why it's done the way it is: we want the tarballs always recreated on "dist". With this patch, that no longer happens.

Of course loops can't be right either, but I can't apply this as-is.

**Comment by pmatilai** [comment]
Just FWIW, https://gitlab.kitware.com/cmake/cmake/-/issues/22852 describes this very case.

**Comment by bracketttc** [comment]
> we want the tarballs always recreated on "dist".

Okay, I get that.

**Comment by pmatilai** [comment]
Solved a bit differently by just dropping the clearly bogus BYPRODUCTS directives: #2900 

Thanks for reporting, and the patch even if it didn't get used as-is!

**Pull request closed without merging**